dp/init.c

```diff
@@ -197,14 +205,120 @@ static inline int port_init(uint8_t port, struct rte_mempool *mbuf_pool)
 	return 0;
 }
 
+#ifdef USE_AF_PACKET
+void
+af_config_network_monitor_interface(uint16_t port_id);
+
+void
+init_af_socks()
+{
+	uint16_t port;
+	struct rte_eth_conf port_conf = port_conf_default;
+	const uint16_t rx_rings = 1, tx_rings = 1;  //tx_rings = rte_lcore_count() - 1;
+	int retval;
+	uint16_t q;
+	char dev_name[LINE_MAX];
+		
+	for (port = S1U_PORT_VETH_ID; port < S1U_PORT_VETH_ID + NUM_SPGW_PORTS; port++) {
+
+		int ifidx;
+		char peer_ifname[IF_NAMESIZE];
+		
+		switch (port) {
+		case S1U_PORT_VETH_ID:
+			ifidx = if_nametoindex(app.ul_iface_name);
+			break;
+		case SGI_PORT_VETH_ID:
+			ifidx = if_nametoindex(app.dl_iface_name);
+			break;
+		default:
+			rte_panic("Unknown port_id: %hu\n", port);
+		}
+
+		if (ifidx == 0)
+			rte_panic("Failed to retrieve ifidx for port: %hu\n", port);
+		/* create full string for net_af_packet */
+		if (if_indextoname(ifidx - 1, peer_ifname) == NULL)
+			rte_panic("Failed to retrieve interface name of peer veth\n");
+		sprintf(dev_name, "net_af_packet%d,iface=%s", port, peer_ifname);
+
+		retval = rte_eth_dev_attach(dev_name, &port);
+
+		if (retval < 0 || port < NUM_SPGW_PORTS)
+			rte_panic("S1U --> dev_str: %s, retval=%d, port_id=%d",
+				  dev_name, retval, port);
+
+#if 0 /* Does not work for dynamic registration of ports */
+		if (port >= rte_eth_dev_count())
+			rte_panic("port (%d) is greater than dev count (%d)!\n",
+				  port, rte_eth_dev_count());
+#endif
+		
+		/* TODO: use q 1 for arp */
```

Comment:
If this is actionable item, briefly describe here, open a corresponding issue, and comment to issue link

Reply:
Deleted